src/Microsoft.Identity.Client/Platforms/uap/UapTokenCacheBlobStorage.cs

```diff
@@ -0,0 +1,106 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.PlatformsCommon.Interfaces;
+using System;
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Threading.Tasks;
+using Windows.Storage;
+using Windows.Storage.Streams;
+
+namespace Microsoft.Identity.Client.Platforms.uap
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <remarks>All continuations must be on the same thread, i.e. ConfigureAwait(true)
+    /// because the TokenCache calls are under lock() so continuing on different threads will cause 
+    /// deadlocks.
+    /// </remarks>
+    internal class UapTokenCacheBlobStorage : ITokenCacheBlobStorage
+    {
+        private const string CacheFileName = "msalcache.dat";
```

Comment:
>"msalcache.dat" [](start = 45, length = 15)

shouldn't this be configurable?

Reply:
We don't really allow users to configure the caching we do for them. They do have the possibility of adding their own TC configuration though.